### PR TITLE
Grant: Identity Wallet and Pallet

### PR DESCRIPTION
**Pull request by JelleMillenaar**
#### Project Abstract

Impierce Technologies proposes the development of a DID Pallet which conforms to the W3C standard and a Digital Identity Wallet that is compatible with EU laws such as eIDAS 2.0 and GDPR and follows internationally recognized standards such as Verifiable Credentials, SIOPv2, and OIDC from the OpenID Foundation. Instead of being specific to only one chain, we would build these components directly for Substrate giving any Substrate-based Blockchain the ability to benefit from Decentralized Identity technology such as passwordless user authentication and verification. 

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by semuelle — COMMENTED** [review]
Thank you for this application, @JelleMillenaar. This is a fairly large proposal, so I'll have to go through it again, but here are some initial questions.

Have you, at any point, considered applying with the Polkadot treasury? I think this project would garner broad enough support and often draws more attention from likeminded people/teams in the space.

File: applications/Identity_wallet_and_pallet.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that e up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We announce the release of the DID Pallet and provide a short explainer of what benefits it brings to chains and encourage them to adopt the Pallet into their Substrate chains. |
+| 1. | Substrate DID Pallet | Release a v1.0 of the DID Pallet with incorporated feedback |
+| 2. | DID Method Specification | Finalized DID Method Specification |
+| 3. | Approval by W3C DID | Approved PR to W3C DID Method Registry for being added |
+| 4. | Integration with Identity.rs | Integrate the DID method into Identity.rs framework |
```
[the remark this answers is not in the document]
Comment (JelleMillenaar) [reply applications/Identity_wallet_and_pallet.md:252]: Identity.rs is technically speaking agnostic to the DLT / DID method, but currently only has IOTA-related implementations. The advantage of using Identity.rs is that it has indeed a full implementation of DID and Verifiable Credentials with a very simple and powerful API. By using Identity.rs, we can ensure any developer that wants to develop another Identity Wallet, Service or Application is protected from accidentally breaking the DID and VC standards or does not follow best practices. 

By using this framework we do not have to implement the Verifiable Credentials standard at all, we only have to provide a front-end (the Wallet) and using it for the implementation of the communication standards (OpenID4VP etc...)

File: applications/Identity_wallet_and_pallet.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+In order to achieve the first phase, we would develop a mature DID Pallet and accompanying Substrate DID Method that fully complies with the DID standard. In addition, we implement a client that can connect to an Identity Provider that is compatible with the SIOPv2, OIDC, OpenID4VP and OpenID4VCI. These are OpenID Foundation standards that allow user authentication following globally recognized standards adopted by hundreds of thousands of websites.
+
+The Identity Wallet is a sensitive piece of software as it will handle private keys and personal information. As such, security is our topmost priority. We have decided to develop the cross-platform app in Rust using the [Tauri framework](https://tauri.app/). It is chosen as it provides the best tools to our developers for separating front-end rendering from business logic and handling sensitive data. This clear separation allows us to handle any sensitive information inside Rust, which gives an extra layer of defense in order to maintain optimal security and privacy.
+
+![Overview of the Components](https://i.imgur.com/VcSkujW.png)
```
[the remark this answers is not in the document]
Comment (JelleMillenaar) [reply applications/Identity_wallet_and_pallet.md:27]: It is indeed simplified. The single lines are actual bi-directional moments of communication that require multiple steps. I just wanted to visualize from a high-level the components and standards and what role they fulfill. It has become easier since your time at uPort. It used to be the wildwest in how an Identity Wallet would communicate with Services, but due to EU legislative developments, OpenID standards have gained a lot of favor and have also matured very nicely. So it is still a significant undertaking, but it is mostly implementing standards. This definitely also favors adoption as most SSI projects will start to shift towards these standards. 

A more detailed diagram would basically be a diagram of the different OpenId message standards, which I have all linked in the document. I can make a more detailed diagram if that helps, but it will purely be a summary of these standards, nothing we would design or architect.

File: applications/Identity_wallet_and_pallet.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+The Substrate / Polkadot and Kusama ecosystems have shown significant interest in Decentralized Identity as there are several identity-related projects, pallets and chains such as the [DID Pallet](https://github.com/substrate-developer-hub/pallet-did), Identity Pallet,  [Parami](https://github.com/parami-foundation/parami-blockchain), [KILT](https://github.com/KILTprotocol/kilt-node) and [Dock](https://github.com/docknetwork/dock-substrate). While the projects are great, there is some major untapped potential. With Decentralized Identity, web3 users can easily authenticate themselves and create trust between each other. This can be immensely valuable to web3 (and web2) service providers, for user authentication and identification. As Impierce Technologies, we do currently warn of the potential of GDPR risks with some of these implementations as writing any Personal Identifiable Information to a Blockchain is most likely a irreversible GDPR violation. With this proposed project we provide a GDPR compliant alternative for the entire Substrate ecosystem.
```
[the remark this answers is not in the document]
Comment (JelleMillenaar) [reply applications/Identity_wallet_and_pallet.md:17]: Some are indeed not GDPR-compliant. I personally highly recommend that no Personal Identifiable Information (PII) should ever be written to a Blockchain, no matter the user consent. This includes names, email addresses, social media handles etc... I have not found such problems with the more SSI focused projects luckily.

**Comment by JelleMillenaar** [comment]
> This is a fairly large proposal, so I'll have to go through it again.

Thanks for the quick initial review @semuelle ! Believe it or not, this is as compact as I could make it.  I am happy to further clarify or deepen any aspect, but I didn't want to overwhelm with details either.

> Have you, at any point, considered applying with the Polkadot treasury? I think this project would garner broad enough support and often draws more attention from likeminded people/teams in the space.

I have spoken to Vincent Chan and Square One for advise on this part. They pointed me to Web3 Grants initially, mostly because we are pretty self-conscious that we are immediately offering a large project (with a relatively large budget) while not having a reputation in the community. As this is reviewed by a more technical committee, we anticipated that it was more likely to focus on the technical details of which we are very confident in our knowledge and experience. Please let me know if this is a misunderstanding on our part, we are still new in the community and trying to find our way ;)

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks a lot for the interesting application and sorry for the delay here. I find it especially interesting that you consider GDPR and have looked into it in detail. Regarding the application, I agree with Sebastian that it’s relatively large at the moment, and we usually like to start with smaller grants and potentially issue follow-up grants. Additionally, we normally focus with these applications on initial PoCs that are potentially useful for others and specifically focus on our ecosystem. So for example Milestone 1-Wallet — UX Designs
 Or Milestone 2-Wallet — OpenID Prototype is not necessarily something that we would fund with our grants program. Whereas Milestone 1-Pallet — MVP DID Pallet is something that we would support, but we would need to have more clarity on how this is different from other existing DID pallets. My recommendation (and not sure you already discussed this with Vincent) would actually be to apply initially for a smaller grant via our grants program to get you started. At the same time, you could potentially start the discussion with the [SBP program](https://substrate.io/ecosystem/substrate-builders-program/) for additional support and/or treasury funding as well as additional support by VCs. I’m also happy to discuss this during a call. Feel free to reach out to david@web3.foundation

**Comment by Noc2** [comment]
Btw. you might remember me, I previously work also on some IOTA projects like Untangle Care  ;-)

**Comment by JelleMillenaar** [comment]
@Noc2 Of course I remember you David! Awesome to see you are still around and working in the Blockchain world :)

I would like to clear up some details on our intent of this application. While we have a bunch of milestones that might not sound very interesting, we honestly just have two projects: Pallet and Wallet. The milestones are just mid-way points so that we can continue to show progress and not have milestones that take 5 months to deliver. So if you want a wallet, you'll need a UX designs and you'll need to implement the OpenID standards. For this request it would be more fitting to review the potential end-result instead of the individual milestones. We can also update the request with fewer but longer milestones if that helps.

> I agree with Sebastian that it’s relatively large at the moment, and we usually like to start with smaller grants and potentially issue follow-up grants

I totally understand that, and we know we are asking for a lot, yet we firmly believe that this project should either be fully accepted or rejected. Within the IOTA ecosystem, we spent so much time building a DID method and simply seeing it fail to gain significant adoption due to a lack of an Identity Wallet. We have 10+ parties from the IOTA community who wanted to use an IOTA wallet, but simply couldn't get funding. I could imagine that your projects like Untangle Care would have benefitted from an Identity Wallet back in the day ;) 

So while I understand you would normally prefer starting with a smaller grant and doing follow up grants, we really believe that the full package is the best outcome for both sides. We can speak from experience that both these pieces are a minimum requirement for an Identity ecosystem, so funding and building only one is not worth your money. If you guys don't think this works for a Web3 grant, but might work as a Polkadot Treasury grant, please let me know and we will apply there instead. 

Thanks again for taking the time to review. I am also happy to further discuss this in a call.

**Comment by Noc2** [comment]
@JelleMillenaar, I just sent you an email to discuss it.

**Comment by JelleMillenaar** [comment]
@semuelle and @Noc2 I have updated the project to fit into a PoC scope. This gives us the ability to proof ourselves to the Polkadot community and built the most important foundational pieces for the Wallet and Pallet. Hopefully this makes the project easier to accept. Our intention is to apply for one follow-up grant after this for the Polkadot Treasury to mature what we have developed here (now with a proven track record), afterwards we should be able to maintain the project on our own resources.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks again for the update. This looks now like it’s a better fit for our grants program. But I have a few follow-up comments:

- Could you better specify as part of the milestone table how your DID pallet will be different from the existing ones? 
- In general, 38,400 USD for a single pallet is a lot, especially if there are already similar existing solutions that you might be able to leverage. I suggest either adding more details or reducing the price. 
- The second milestone ideally should also contain technical deliveries as part of the table that is specific to our ecosystem. Otherwise, we risk funding a wallet for something unrelated to the Polkadot/Kusama ecosystem.

**Comment by keeganquigley** [comment]
@JelleMillenaar thanks for the application and the updates.

- I agree that the price is currently too high for a single DID pallet. I also personally think that M2 price is too high for a wallet front-end. Would you be willing to lower the price of either milestone?
- I like that Tauri is written in Rust!
- My main concern with creating another identity wallet is that it might be hard to get users to adopt it. For example, I'm not sure if Kilt's Sporran wallet has gained much traction yet at this point. Regarding features, can you give more insight as to how your wallet will differ from Sporran?
- Regarding the additional funding you received, this is for work that is not related to the grant deliverables, correct? Just making sure, as we don't want to double-fund for the same work.

File: applications/Identity_wallet_and_pallet.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 |
+| **0b.** | Documentation | We will provide **inline documentation** of the code, contribution guidelines for translations and a white labeling guide. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests. We will set up basic end-to-end testing and perform user testing. |
+| **0d.** | Docker | Where relevant and possible, we will deliver a Dockerfile to test out the functionality of this milestone. |
+| 0e. | Article | We will publish an **article** that announced the release of the app and how it can be used or how parties may integrate with this new technology! |
+| 1. | Identity Wallet |  The Identity Wallet repo is updated with a MVP Tauri Identity Wallet including a standard design. |
+| 2. | Identity Client | We deliver an Identity Client that supports SIOPv2. |
```
Comment (keeganquigley) [inline applications/Identity_wallet_and_pallet.md:189]: What do you mean by "identity client"? How is this different from the wallet?

**Comment by JelleMillenaar** [comment]
@Noc2 
> Could you better specify as part of the milestone table how your DID pallet will be different from the existing ones?

I have updated the milestone description with a small overview. Unfortunately, the Pallet isn't a simple copy-paste from existing implementations. While the DID standard is relatively easy to understand and implement on the base-level, full compliance is challenging due to the (bit odd) controllers feature, which allows 3rd party managed keys to update the state of another DID Document. It is a level of delegation of control that requires excellent design and extensive edge case testing, especially since we are working in a decentralized system that can trigger race conditions. A completely valid DID update can become invalid due to consensus level ordering. In addition, we want to reduce fees by compressing the DID update messages, yet keep it future proof if any compression algorithm needs to be adjusted in the future. 

> In general, 38,400 USD for a single pallet is a lot, especially if there are already similar existing solutions that you might be able to leverage. I suggest either adding more details or reducing the price.

Overall we believe that the Pallet is a large project and we wouldn't feel comfortable with lowering the price as we would do this development with good software practices such as code reviewing, testing, documentation and designing and writing a DID Method Specification, while we also would like to engage with the wider community for testing, feedback and achieving Pallet adoption. 

> The second milestone ideally should also contain technical deliveries as part of the table that is specific to our ecosystem. Otherwise, we risk funding a wallet for something unrelated to the Polkadot/Kusama ecosystem.

Excellent point, I have added that the Wallet will specifically use the DID Pallet we deliver in the first milestone. 

@keeganquigley Thanks for taking a look as well!

> I also personally think that M2 price is too high for a wallet front-end. 

I think you misunderstood a bit, Tauri is just a framework, we will still need to implement our own "back-end" for the wallet. All business logic such as state management, secret management, cryptographic operations for encryption at rest and all operations for creating, verifying and receiving Verifiable Credentials will need to be written and developed by us in Rust. Our front-end will be limited to carry close-to-zero business logic in order to keep everything clean and secure inside the Rust environment. The age of Tauri makes it a bit more tricky as Tauri is only in alpha stage for mobile development, so we will also likely contribute feedback and even fixes to Tauri along the way. We have tested the alpha version and are certain that all minimum required features for a mobile framework are available to us to develop this app.  

> My main concern with creating another identity wallet is that it might be hard to get users to adopt it. For example, I'm not sure if Kilt's Sporran wallet has gained much traction yet at this point. 

Honestly an excellent question. We see an explosion of Identity Wallet across different ecosystems, often without leading to much adoption. We have several reasons as to why we believe to be different.

1. We are not opinionated on a single blockchain / token. If all 150 Substrate-based blockchains adopt the DID pallet, our single app will be able to serve them all. That is why we targeted Substrate, because we see a lot of potential of bringing DID functionalities to a collection of ecosystems, rather then an individual project. Naturally not all blockchains will need to adopt the DID Pallet, as long as projects within a certain ecosystem are happy to use a particular blockchain with the Pallet for validation, our app will be available to them.
2. We focus on creating instant daily demand by bridging the old and the new world by adopting OpenID standards. With a vibrant community such as from Substrate / Polkadot, many websites could easily add a Login with Substrate functionality to their website, allowing users to not just log-in, but easily share information with these websites without registration and without requiring the website to store this information.
3. Impierce Technologies offers eIDAS 2.0 compliant services to web2 companies, government agencies and web3 projects to easily onboard into this SSI ecosystem and benefit from users bringing their own identities, instead of account creation. We focus on creating a network effect in a few industries such that users can start using the wallet for their daily interactions with these companies / services. 
4. We focus on security and excellent UX design. We hide all technological mumbo-jumbo and focus on creating a user experience that enhances the current browsing experience.
5. Our future plans of the wallet with the extension system allows any community, company, DAO or government to introduce necessary features into the wallet to engage with their ecosystem, connecting the ecosystems into one wallet.

TL;DR We focus on connecting multiple ecosystems instead of making our own little one. 

> Regarding the additional funding you received, this is for work that is not related to the grant deliverables, correct? Just making sure, as we don't want to double-fund for the same work.

The other funding received will help extend the scope of this wallet, but indeed will not be spent on any of the features or milestones listed in this proposal.

**Comment by keeganquigley** [comment]
Thanks @JelleMillenaar I will go ahead and mark your application as ready for review so the other committee members can take a look. However, I personally think the price is still a bit high for me to approve at this point.

**Comment by JelleMillenaar** [comment]
Thanks @keeganquigley I appreciate it moving forward. Perhaps I can provide some final information on the budget. We are a Dutch / Germany based development team, which creates above average salary requirements. Even after we shaped this project as a PoC, code quality and security are important to us. So we plan to deliver a very high-quality solution which also contains a lot of technical challenges such as bridging the SIOP and OIDC standards, which isn't common practice. Overall we have estimated the full development of an Identity Wallet (without implementing a DID method) to cost us around €250.000 as we have a lot of additional features planned and want to design very secure and high-quality open-source software which probably will be an investment of 1.5 years of development with a sizeable team (This contains multiple releases so it won't take that long before it becomes useful). This is of course completely out-of-scope of what we ask here, but perhaps it grounds our calculations a bit better as this is an initial PoC towards that project. 

Lastly, we have experience in designing and implementing the did:iota method, which costed much more time and money as we planned here. Luckily we can leverage our experience to do it much faster this time around ;) 

I appreciate that your team is careful to not throw around large amounts of money for no reason, so we will respect a rejection based on the budget being too high, but unfortunately we find it difficult to find a way to lower our costs. Feel free to ask any more questions if they come up and thanks again for your time!

**Comment by randombishop** [comment]
Hi @JelleMillenaar and thank you for this application. I found it very interesting and instructive.

I have a couple of questions:

1. Why would Kilt be non GDPR compliant. To my knowledge, the default is for users to only store hashes so the data doesn't actually live in the blockchain, but just crypto proofs. I also believe future versions will even support zero knowledge selective information proofs so that would be technically zero data. Can you please elaborate on how exactly the current projects are non compliant.

2. You propose the use of Tauri framework and invoke security benefits. I am not familiar with it so can you please elaborate on how using it improves security in comparison with other more common front-end frameworks? Is it simply because Rust is harder to reverse engineer or are there other principles and/or technologies involved?

**Comment by JelleMillenaar** [comment]
@randombishop Thank you for your response. 

> Can you please elaborate on how exactly the current projects are non compliant.

My biggest concern lies not with the DID pallets such as from KILT, as the SSI community is very focused on GDPR compliance. However, the Identity Pallet used for the [RFP: Substrate Identity Directory](https://github.com/w3f/Grants-Program/blob/master/docs/RFPs/Under%20Development/identity-directory.md) states "In Substrate-based chains which implement the [Identity pallet](https://github.com/paritytech/substrate/tree/master/frame/identity), users can register on-chain information about themselves." which is highly concerning. 

That being said, the KILT implementation does put hashes of personal identifiable information on the blockchain. Unfortunately, it is currently unclear if that is considered GDPR compliant., and we prefer to err on the side of caution. More strict GDPR interpretations have stated that hashing does not fully anonymize data, no matter how salted. While I don't agree with the interpretation as I don't believe it adds a security risk, the law is not written by developers. As a result, the SSI community commonly avoids putting hashes of PII on the chain. 

The biggest reason why I would recommend KILT to remove the hash is that adding the hash on the chain also does not achieve anything. The verification of a Verifiable Credential is done through signatures, signed by keys associated with DIDs. The DID Documents that contain the keys need to be uploaded to a chain in order to verify the signature without having to contact the Issuer. Some might argue that DLT adds a verifiable timestamp to the issuance of the credential. The Verifiable Credential is able to carry a timestamp, which is written and signed by the Issuer. Since the Verifier already trusts the Issuer to be a source of the information inside the VC, it naturally would also trust the timestamp. Therefore, most SSI solutions do not upload any hash of the VC to a chain. Eventually, Revocation is another topic, but that is out-of-scope for this discussion.

Lastly, I would want to add that we don't argue that the Substrate ecosystem has no real SSI solutions and that we are the only one able to deliver that as that is simply not true. We do think we have an elegant solution in mind to bring SSI capabilities to many more chains without them having to invest in developing those parts, nor do they have to interact with a separate dedicated SSI chain such as KILT. 

> can you please elaborate on how using it improves security in comparison with other more common front-end frameworks? 

Tauri is not a front-end framework, but rather a mobile back-end framework similar to what Electron is for Desktop applications. It allows us to completely isolate the front-end rendering logic (in unsecured Javascript) from the business logic in Rust ([Isolation Pattern](https://tauri.app/v1/references/architecture/inter-process-communication/isolation/)). Rust is a language that results in much fewer vulnerabilities in the code, directly increasing security. Tauri is built in such a way as to leverage that as much as possible by inherently supporting the isolation pattern.  As all the sensitive data in the application is purely handled in Rust, we believe that adds additional layers of defense. It's just one piece of our overall security strategy, but very important already. An amazing side-effect of using Tauri is that their application sizes are significantly smaller, which reduces download time and storage requirements. We have also worked with the Tauri developers and know their work ethics and high focus on quality and security, making it a perfect fit. With 62k stars on Github, we aren't the only ones being positive about it ;)

**Comment by randombishop** [comment]
@JelleMillenaar Thank you for your feedback. 

My opinion on this application is that there are already several decentralized identity initiatives in the polkadot ecosystem; and a better course of action would be to build/improve on top of an existing one rather than build a new one. It's also still unclear how others are compliant or not with different regulations so can't really assess if this version would be more compliant.

Therefore I won't approve it; but my opinion being subjective, let's keep the application open for the rest of the team to review.

File: applications/Identity_wallet_and_pallet.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+	* Daniel Mader - Full-stack / Rust Developer
+	* Currently Hiring a Mobile Developer
+
+If this project is accepted, we would hire an additional Rust developer with Substrate experience and a part-time front-end developer.
+
+### Contact
+
+- **Contact Name:** Jelle Millenaar
+- **Contact Email:** Jelle.millenaar@impierce.com
+- **Website:** [impierce.com](impierce.com)
```
Comment (semuelle) [inline applications/Identity_wallet_and_pallet.md:75]:
```suggestion
- **Website:** [impierce.com](https://impierce.com)

```

**Review by semuelle — APPROVED** [review]
Sorry for not getting back to you sooner, @JelleMillenaar. I'm happy to approve the project in its current form. As @Noc2 mentioned, it's fairly expensive for what it is, but if you are able to implement your entire roadmap, this project could be very valuable for the ecosystem. Also, building a pallet for others to pick up might not be the best route, but there might be enough incentive to do so if everything else (W3C standards implementation, wallet) is in place.

**Review by takahser — CHANGES_REQUESTED** [review]
@JelleMillenaar I appreciate your diligent work in developing this proposal. I'd like to share a few thoughts:

- Regarding to the deliverable _3. DID Testnet_ under Milestone 1 - we usually don't support deployment tasks. Hence, you are free to exclude it.
- Considering the scaled-back M1, would you be willing to apply for a Level 2 grant for M1 only? I'd be willing to approve it, taking into account the scope, the grant amount (although still rather expensive), and your first-time grantee status.
- While you're transitioning from the IOTA ecosystem, I'm curious about your choice of Substrate. I support this decision, I'm simply interested in the rationale. There are no wrong answers here. ;)

**Comment by JelleMillenaar** [comment]
Thanks for your questions and the reviews. We have decided to close this request for now as we have found another grant in another ecosystem. We continue to have interest in Substrate and are planning for a multi-DLT set-up, so we might reapply later. But we do want to focus on one ecosystem first. That reapplication will assume the more fundamental work from our side will be done, so it will likely also fit better based on your comments and feedback. Thanks again for consideration and hopefully we will submit something more exciting in a while!

**Comment by semuelle** [comment]
I'm glad you found funding, but I do hope you'll reapply later, @JelleMillenaar. Best of luck!
